subsys/mgmt/mcumgr/grp/img_mgmt/src/img_mgmt.c

```diff
@@ -537,15 +541,34 @@ img_mgmt_upload(struct smp_streamer *ctxt)
 
 		if (g_img_mgmt_state.off == g_img_mgmt_state.size) {
 			/* Done */
+			reset = true;
+
+#ifdef CONFIG_IMG_ENABLE_IMAGE_CHECK
+			static struct flash_img_context ctx;
+
+			if (flash_img_init_id(&ctx, g_img_mgmt_state.area_id) == 0) {
+				struct flash_img_check fic = {
+					.match = g_img_mgmt_state.data_sha;
```

Comment:
I do not think we can reuse the "sha" value here, as defined in https://docs.zephyrproject.org/latest/services/device_mgmt/smp_groups/smp_group_1.html#mcumgr-smp-group-1.
This is used for session identification but can be any value, for example when you want to restart upload of the same image completely, you would just use different value here than in previous session.

Reply:
We should update that document to state that it is an sha256 of the image and nothing else is accepted, it makes no sense to have it as any possible value. To restart from the beginning, the erase command should be used.

Reply:
> We should update that document to state that it is an sha256 of the image and nothing else is accepted, it makes no sense to have it as any possible value. To restart from the beginning, the erase command should be used.

No. Because we have stream flash and progressive erase which should allow to upload without pre-erasing slot.
Forcing erase would break these features.